src/main/scala/beam/sim/BeamHelper.scala

```diff
@@ -460,7 +478,31 @@ trait BeamHelper extends LazyLogging {
     run(beamServices)
   }
 
-  def setupBeamWithConfig(config: TypesafeConfig): (MutableScenario, String, NetworkCoordinator) = {
+  private def fillScenarioFromExternalSources(
+    injector: inject.Injector,
+    matsimConfig: Config,
+    networkCoordinator: NetworkCoordinator,
+    beamServices: BeamServices
+  ): Scenario = {
+    val beamConfig = beamServices.beamConfig
+    val useExternalDataForScenario: Boolean =
+      Option(beamConfig.beam.exchange.scenario.folder).exists(!_.isEmpty)
+    val matsimScenario = buildScenarioFromMatsimConfig(matsimConfig, networkCoordinator)
+
+    if (useExternalDataForScenario) {
+      val scenarioSource: ScenarioSource = getScenarioSource2(injector, beamConfig)
+      ProfilingUtils.timed(s"Load scenario using ${scenarioSource.getClass}", x => logger.info(x)) {
+        new ScenarioLoader(matsimScenario, beamServices, scenarioSource).loadScenario()
+        matsimScenario
+      }
+    } else {
+      matsimScenario
+    }
+  }
+
+  def setupBeamWithConfig(
+    config: TypesafeConfig
+  ): (BeamConfig, org.matsim.core.config.Config, String, NetworkCoordinator) = {
```

Comment:
Let's extract it to case class return type. Should be easier to read and maintain